src/WebJobs.Extensions.Twilio/Config/TwilioJobHostConfigurationExtensions.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs.Extensions.Bindings;
+using Microsoft.Azure.WebJobs.Extensions.Twilio;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Newtonsoft.Json.Linq;
+using Twilio;
+
+namespace Microsoft.Azure.WebJobs
+{
+    /// <summary>
+    /// Extension methods for Twilio integration
+    /// </summary>
+    public static class TwilioJobHostConfigurationExtensions
+    {
+        /// <summary>
+        /// Enables use of the Twilio extensions
+        /// </summary>
+        /// <param name="config">The <see cref="JobHostConfiguration"/> to configure.</param>
+        /// <param name="twilioSmsConfig">The <see cref="TwilioSmsConfiguration"/> to use.</param>
+        public static void UseTwilioSms(this JobHostConfiguration config, TwilioSmsConfiguration twilioSmsConfig = null)
+        {
+            if (config == null)
+            {
+                throw new ArgumentNullException("config");
+            }
+
+            if (twilioSmsConfig == null)
+            {
+                twilioSmsConfig = new TwilioSmsConfiguration();
+            }
+
+            config.RegisterExtensionConfigProvider(new TwilioSmsConfig(twilioSmsConfig));
+        }
+
+        internal class TwilioSmsConfig : IExtensionConfigProvider
+        {
+            private TwilioSmsConfiguration _twilioConfig;
+            private TwilioRestClient _twilioClient;
+
+            public TwilioSmsConfig(TwilioSmsConfiguration twilioConfig)
+            {
+                _twilioConfig = twilioConfig;
+            }
+
+            public void Initialize(ExtensionConfigContext context)
+            {
+                if (context == null)
+                {
+                    throw new ArgumentNullException("context");
+                }
+
+                if (string.IsNullOrEmpty(_twilioConfig.AccountSid))
+                {
+                    ThrowMissingSettingException("AccountSID", TwilioSmsConfiguration.AzureWebJobsTwilioAccountSidKeyName, "AccountSID");
+                }
+
+                if (string.IsNullOrEmpty(_twilioConfig.AuthToken))
+                {
+                    ThrowMissingSettingException("AuthToken", TwilioSmsConfiguration.AzureWebJobsTwilioAccountAuthTokenName, "AuthToken");
+                }
+
+                _twilioClient = new TwilioRestClient(_twilioConfig.AccountSid, _twilioConfig.AuthToken);
```

Comment:
I could certainly keep things as is, but if the effort is not significant, I think it's worth spending a bit more time to get this working as Brett is describing, since there are some scenarios (e.g. test tokens) that would benefit from it.